src/wp-includes/canonical.php

```diff
@@ -740,6 +754,30 @@ function lowercase_octets( $matches ) {
 		$requested_url = preg_replace_callback( '|%[a-fA-F0-9][a-fA-F0-9]|', 'lowercase_octets', $requested_url );
 	}
 
+	if (
+		$redirect_obj &&
+		is_a( $redirect_obj, 'WP_Post' )
+	) {
```

Comment:
```suggestion
	if ( $redirect_obj instanceof WP_Post ) {
```

From looking thru Core usages, I think `instanceof` is the preferred pattern.